utils/src/lib.rs

```diff
@@ -262,12 +266,23 @@ fn recurse_link<'a>(
         dependencies.dependencies.iter().for_each(|dep| {
             let ArtifactDependency {  file, key, .. } = dep;
             let next_target = format!("{file}:{key}");
+            let root = PathBuf::from(root.as_ref().to_str().unwrap());
+            // In some project setups, like JS-style workspaces, you might not have node_modules available at the root of the foundry project.
+            // In this case, imported dependencies from outside the root might not have their paths tripped correctly.
+            // Therefore, we fall back to a manual path join to locate the file.
+            let fallback_path =  dunce::canonicalize(root.join(file)).unwrap_or_else(|_| panic!("No file named {file}"));
```

Comment:
I don't like that we panic here,

when does this return None?

Reply:
It returns none if the path doesn't exist or is not a directory (and other OS related errors in the case of windows, probably): https://doc.rust-lang.org/std/fs/fn.canonicalize.html

moved all the fallback code on the error case if the artifact is not found—if we're gonna panic anyway there's no way to stop it at this point (as it's an unhandled case / project setup). And yeah we need to rewrite this 😅